### DISCUSSION TITLE
I want to get message set for a chat using the remoteJid?

### DISCUSSION THREAD

**Discussion by saadnust** [WhiskeySockets/Baileys#311] [General Questions]
Looking to fetch all/recent messages of particular chat using its remoteJid, on login I get a message set but that's not complete and not all chats are available in it. So I want to search chats for messages using Jid

**Comment by joseorlandofp** [comment]
I believe that is not possible. You have to set syncFullHistory to true upon socket initialization and store yourself the messages that come in the socket.ev.on('messages-upsert') and then query your database.

```ts
 const socket = makeWASocket({
     shouldSyncHistoryMessage: (msg: proto.Message.IHistorySyncNotification) => true,
    syncFullHistory: false,
    fireInitQueries: true
    ...
  });

```
